tokio/src/lib.rs

```diff
@@ -463,12 +463,18 @@ compile_error! {
         feature = "io-std",
         feature = "net",
         feature = "process",
-        feature = "rt-multi-thread",
         feature = "signal"
     )
 ))]
 compile_error!("Only features sync,macros,io-util,rt,time are supported on wasm.");
 
+#[cfg(all(
+    not(tokio_unstable),
+    all(target_family = "wasm", not(target_os = "wasi")),
+    feature = "rt-multi-thread"
+))]
+compile_error!("rt-multi-thread feature only available on wasi targets.");
```

Comment:
It's possible I'm misreading these conditions, but as far as I can tell, this has the effect of making rt-multi-thread work on wasi _without_ the `--cfg tokio_unstable`. I don't want to do that in the same PR as the one that adds the initial support for it.